slice.go

```diff
@@ -693,3 +694,64 @@ func Splice[T any, Slice ~[]T](collection Slice, i int, elements ...T) Slice {
 
 	return append(append(append(output, collection[:i]...), elements...), collection[i:]...)
 }
+
+// Pull removes all given values from slice using SameValueZero for equality comparisons.
+func Pull[T comparable, Slice ~[]T](collection Slice, elements ...T) Slice {
+	output := make(Slice, 0, len(collection))
+
+	for _, item := range collection {
+		if !slices.Contains(elements, item) {
+			output = append(output, item)
+		}
+	}
+
+	return output
+}
+
+// PullAllBy is like [Pull] except that it accepts iteratee which is invoked
+// for each element of slice and values to generate the criterion by which they're compared
+func PullAllBy[T any, U comparable, Slice ~[]T](collection Slice, iteratee func(T) U, elements ...U) Slice {
+	elementCriteria := make(map[U]struct{}, len(elements))
+	for _, element := range elements {
+		elementCriteria[element] = struct{}{}
+	}
+
+	var output Slice
+	for _, item := range collection {
+		if _, shouldRemove := elementCriteria[iteratee(item)]; !shouldRemove {
+			output = append(output, item)
+		}
+	}
+
+	return output
+}
+
+// PullAt removes elements from slice corresponding to indexes and returns
+// an array of removed elements
+func PullAt[T any](slice []T, indexes ...int) ([]T, []T) {
+	if len(slice) == 0 || len(indexes) == 0 {
+		return nil, slice
+	}
+
+	sort.Sort(sort.Reverse(sort.IntSlice(indexes)))
+
+	indexMap := make(map[int]struct{}, len(indexes))
```

Comment:
Please check and test the behavior if you pass multiple times the same index in `indexes`

I'm worried because of the code a few lines below

```go
output := make([]T, 0, len(slice)-len(indexMap))
```

Imagine a slice with 1 item and pass 0,1,2 as indexes, or 0,0,0

One solution could be to call slices.Compact on indexes before creating indexMap

Reply:
Thank you for your comment, I has checked first case and described it in tests, that was working correctly.
Second case working correctly too, added test cases for both